src/components/MainHome.tsx

```diff
@@ -0,0 +1,109 @@
+import React, { Component } from 'react';
+import { RouteComponentProps } from 'react-router-dom';
+import './MainHome.css';
+import LeftTabBar from './LeftTabBar';
+import CustomButton from './CustomButton';
+import Room from './Room';
+import openSocket from 'socket.io-client';
+import Constants from '../Constants';
+import * as Types from '../Types';
+import CustomTextInput from './CustomTextInput';
+import TrieSearch from 'trie-search';
+
+type MainHomeState = {
+    sendTo: string;
+    connectToRoom: boolean;
+    roomid: string;
+    socket: SocketIOClient.Socket;
+    displayName: Types.UserDisplay;
+    users: Types.UserDisplayMap;
+    searchResults: Array<Types.UserDisplay>;
+};
+
+let usersTrie: Record<string, any>;
+
+export default class MainHome extends Component<RouteComponentProps, MainHomeState> {
+    state: MainHomeState = {
+        sendTo: '',
+        connectToRoom: false,
+        roomid: '',
+        socket: openSocket(`${Constants.SERVER_HOST}:${Constants.SERVER_PORT}`),
```

Comment:
So is all the socket logic tied to this component? Do you think it'd be helpful if we split out the socket connection to a separate module and then use that module inside of this component?

Reply:
Since we're passing this socket to other components, and we don't need to open connections other than this one, I feel like it would be beneficial to move the socket out to some other centralized place instead of directly on a component.

Reply:
Yea that makes sense, would it be something like this:

[How to use a single instance of socket-io](https://itnext.io/how-to-use-a-single-instance-of-socket-io-in-your-react-app-6a4465dcb398)

I still like having the socket.on messages in each component because we can change the state accordingly right away.

Reply:
Ya, the context thing looks good for it